service/storage_service.cc

```diff
@@ -6936,6 +6936,61 @@ future<> storage_service::move_tablet(table_id table, dht::token token, locator:
     });
 }
 
+future<locator::load_stats> storage_service::load_stats_for_tablet_based_tables() {
+    auto holder = _async_gate.hold();
+
+    if (this_shard_id() != 0) {
+        // group0 is only set on shard 0.
+        co_return co_await container().invoke_on(0, [&] (auto& ss) {
+            return ss.load_stats_for_tablet_based_tables();
+        });
+    }
+
+    auto load_stats = co_await _db.map_reduce0([] (replica::database& db) mutable -> future<locator::load_stats> {
+        locator::load_stats load_stats{};
+        auto& tables_metadata = db.get_tables_metadata();
+
+        co_await tables_metadata.for_each_table_gently([&] (table_id id, lw_shared_ptr<replica::table> table) -> future<> {
+            auto erm = table->get_effective_replication_map();
+            auto& token_metadata = erm->get_token_metadata();
+            auto& tmap = token_metadata.tablets().get_tablet_map(id);
+            auto me = locator::tablet_replica { token_metadata.get_my_id(), this_shard_id() };
+
+            // It's important to tackle the anomaly in reported size, since both leaving and
+            // pending replicas could otherwise be accounted during tablet migration.
+            // If transition hasn't reached cleanup stage, then leaving replicas are accounted.
+            // If transition is past cleanup stage, then pending replicas are accounted.
+            // This helps to reduce the discrepancy window.
+            auto tablet_filter = [&tmap, &me] (locator::global_tablet_id id) {
+                auto transition = tmap.get_tablet_transition_info(id.tablet);
+                auto& info = tmap.get_tablet_info(id.tablet);
+
+                // if tablet is not in transit, it's filtered in.
+                if (!transition) {
+                    return true;
+                }
+
+                bool is_pending = transition->pending_replica == me;
+                bool is_leaving = locator::get_leaving_replica(info, *transition) == me;
+                auto s = transition->stage;
+
+                return (!is_pending && !is_leaving)
+                       || (is_leaving && s < locator::tablet_transition_stage::cleanup)
+                       || (is_pending && s >= locator::tablet_transition_stage::cleanup);
```

Comment:
Note: we can, still, double-account since the RPC will hit the node at different times.

This isn't a problem with a global tablet map. But we can ignore it for now.

Reply:
Exactly, a few days ago, I raised it in a conversation with @tgrabiec and we decided it's good enough and discrepancy shouldn't last long.